docs/csharp/nullable-attributes.md

```diff
@@ -0,0 +1,276 @@
+
+Which of these strategies you pick depends on how much active development is taking place in your project. The more mature and stable your project, the better the second strategy. The more active features are being developed, the better the first strategy.
+
+## Should nullable warnings introduce breaking changes?
+
+Before you enable nullable reference types, variables are considered *nullable oblivious*. Once you enable nullable reference types, all those variables are *non-nullable*. The compiler will issue warnings if those variables aren't initialized to non-null values.
+
+Another likely source of warnings is return values when the value hasn't been initialized.
+
+The first step is to use `?` annotations on parameters and return types to indicate when arguments or return values may be null, or must not be null. As you do this, your goal isn't just to fix warnings. The more important goal is to make the compiler understand your intent for potential null values. As you examine the warnings, you reach your next major decision for your library. Do you want to consider modifying API signatures to more clearly communicate your design intent? A better API signature for the `TryGetMessage` method examined earlier could be:
+
+
+```csharp
+string? TryGetMessage(string key);
+```
+
+The return value indicates success or failure, and carries the value if the value was found. In many cases, changing API signatures can improve how they communicate null values.
+
+However, for public libraries, or libraries with large user bases, you may prefer not introducing any API signature changes. For those cases, and other common patterns, you can apply attributes to more clearly define when an argument or return value may be `null`. Whether or not you consider changing the surface of your API, you'll likely find that type annotations alone aren't sufficient for describing when `null` values for arguments or return values. In those instances, you can apply attributes to more clearly describe an API. 
+
+## Attributes extend type annotations
+
+several attributes have been added to express further information about the null state of variables. All code you wrote before C# 8 introduced nullable reference types was *null oblivious*. That means any reference type variable may be null, but null checks aren't required. Once your code is *nullable aware*, those rules change. Reference types should never be the `null` value, and nullable reference types must be checked against null before being dereferenced.
+
+The rules for your APIs are likely more complicated, as you saw wit the `TryGetValue` API scenario. Many of your APIs have more complex rules for when variables can or can't be `null`. In these cases, you'll use one of these attributes to express those rules:
+
+- `AllowNull`: A non-nullable input argument may be null.
```

Comment:
It might be worth mentioning that these descriptions may sound odd at first, but will be explained later on. Since the rationale for their behavior is pretty complicated, it may even be better to recap each kind of attribute "family" with this at the end.

Reply:
Great suggestion